test/snapshot_collection_test.go

```diff
@@ -0,0 +1,101 @@
+	suite.Suite
+	groupService              *service.NsGroupService
+	snapcollService           *service.SnapshotCollectionService
+	volumeService             *service.VolumeService
+	volcollService            *service.VolumeCollectionService
+	protectionScheduleService *service.ProtectionScheduleService
+}
+
+func (suite *SnapCollWorkflowSuite) SetupTest() {
+	groupService, err := config()
+	assert.Nilf(suite.T(), err, "Unable to connect to group, err: %v", err)
+	suite.groupService = groupService
+	suite.snapcollService = groupService.GetSnapshotCollectionService()
+	suite.volumeService = groupService.GetVolumeService()
+	suite.volcollService = groupService.GetVolumeCollectionService()
+	suite.protectionScheduleService = groupService.GetProtectionScheduleService()
+	_, err = createDefaultVolume(suite.volumeService)
+	assert.Nilf(suite.T(), err, "Unable to create default volume, err: %v", err)
+	_, err = createDefaultVolColl(suite.volcollService)
+	assert.Nilf(suite.T(), err, "Unable to create default volume collection, err: %v", err)
+}
+
+func (suite *SnapCollWorkflowSuite) TearDownSuite() {
+	err := deleteDefaultVolume(suite.volumeService)
+	assert.Nilf(suite.T(), err, "Unable to delete default volume, err: %v", err)
+	err = deleteDefaultVolColl(suite.volcollService)
+	assert.Nilf(suite.T(), err, "Unable to delete default volume collection, err: %v", err)
+	suite.deleteSnapColl(snapcollName)
+}
+
+func (suite *SnapCollWorkflowSuite) deleteSnapColl(snapcollName string) {
+	snapcoll, _ := suite.snapcollService.GetSnapshotCollectionByName(snapcollName)
+	if snapcoll != nil {
+		err := suite.snapcollService.DeleteSnapshotCollection(*snapcoll.ID)
+		assert.Nilf(suite.T(), err, "Unable to delete snapshot collection, err: %v", err)
+	}
+}
+
+func (suite *SnapCollWorkflowSuite) TestCreateSnapshotCollection() {
+	// Create volume collection & associate volumes
+	volcoll, _ := suite.volcollService.GetVolumeCollectionByName(defaultVolCollName)
+	vol, _ := suite.volumeService.GetVolumeByName(defaultVolumeName)
+	err := suite.volumeService.AssociateVolume(*vol.ID, *volcoll.ID)
+	assert.Nilf(suite.T(), err, "Unable to associate volume to volume collection, err: %v", err)
+	// Create snapshot schedule and set frequency to 1 min
+	var period *int64 = param.NewInt64(1)
+	var numRetain *int64 = param.NewInt64(12)
+	newPS := &nimbleos.ProtectionSchedule{
+		Name:                  param.NewString(protectionScheduleName),
+		VolcollOrProttmplType: nimbleos.NsProtectionPolicyTypeVolumeCollection,
+		VolcollOrProttmplId:   volcoll.ID,
+		Period:                period,
+		PeriodUnit:            nimbleos.NsPeriodUnitMinutes,
+		NumRetain:             numRetain,
+	}
+	_, err = suite.protectionScheduleService.CreateProtectionSchedule(newPS)
+	assert.Nilf(suite.T(), err, "Unable to create protection schedule", err)
+
+	filter := &param.GetParams{}
+	snapcollBefore, _ := suite.snapcollService.GetSnapshotCollections(filter)
+
+	time.Sleep(60 * time.Second)
```

Comment:
It is unfortunate to insert this sort of delay in a test run.  I see it is probably unavoidable given the constraints of what is being tested.  However...

I still do not have an understanding of the purpose of these tests, despite asking several times.  If the intent is to test the behavior of the storage system, I'll ask why such tests belong in this project.  If the intent is to test the behavior of the Go client SDK, I'll ask why we need to verify the storage system is properly creating snapshot collections.  Wouldn't we simply verify that the protection schedule was created as requested?

Reply:
I am sorry I missed replying to your question earlier. Intent is to test Go client sdk. The test case was written as such to make sure the snapshot collection is automatically triggered. But what you are saying makes sense to me. I will still confirm it with Vidhut and remove this check if its not required.